src/context/GameLogic.tsx

```diff
@@ -31,23 +27,39 @@ export const useGameDetails = () => {
 // Context provider that returns the context that wraps the children to give them access to the state
 export const GameContextProvider = ({ children }: any) => {
   const [deck, setDeck] = useState<Card[]>(initDeck());
-  const [selectedCards, setSelectedCards] = useState<Card[]>([]);
 
   const value = {
     deck,
     setDeck,
-    selectedCards,
-    setSelectedCards,
   };
 
   useEffect(() => {
+    const selectedCards = deck.filter((card) => card.selected === true);
```

Comment:
I know we'll revisit this in upcoming work! let's keep some 👀  on it. We can rework some of the logic